pkg/cmd/pr/create/create.go

```diff
@@ -127,23 +145,239 @@ func NewCmdCreate(f *cmdutil.Factory, runF func(*CreateOptions) error) *cobra.Co
 	return cmd
 }
 
-func createRun(opts *CreateOptions) error {
+func createRun(opts *CreateOptions) (err error) {
+	httpClient, err := opts.HttpClient()
+	if err != nil {
+		return err
+	}
+	client := api.NewClientFromHTTP(httpClient)
+
+	ctx, err := NewCreateContext(opts)
+	if err != nil {
+		return err
+	}
+
+	var milestoneTitles []string
+	if opts.Milestone != "" {
+		milestoneTitles = []string{opts.Milestone}
+	}
+
+	state := shared.IssueMetadataState{
+		Type:       shared.PRMetadata,
+		Reviewers:  opts.Reviewers,
+		Assignees:  opts.Assignees,
+		Labels:     opts.Labels,
+		Projects:   opts.Projects,
+		Milestones: milestoneTitles,
+	}
+
+	defaultsErr := computeDefaults(*ctx, &state)
+	if defaultsErr != nil && (opts.Autofill || opts.WebMode || !opts.Interactive) {
+		return fmt.Errorf("could not compute title or body defaults: %w", defaultsErr)
+	}
+
+	if opts.TitleProvided {
+		state.Title = opts.Title
+	}
+
+	if opts.BodyProvided {
+		state.Body = opts.Body
+	}
+
+	if opts.WebMode {
+		err := handlePush(*opts, *ctx)
+		if err != nil {
+			return err
+		}
+		return previewPR(*opts, *ctx, state)
+	}
+
+	if opts.Autofill || !opts.Interactive {
+		return submitPR(*opts, *ctx, state)
+	}
+
+	existingPR, err := api.PullRequestForBranch(
+		client, ctx.BaseRepo, ctx.BaseBranch, ctx.HeadBranchLabel, []string{"OPEN"})
+	var notFound *api.NotFoundError
+	if err != nil && !errors.As(err, &notFound) {
+		return fmt.Errorf("error checking for existing pull request: %w", err)
+	}
+	if err == nil {
+		return fmt.Errorf("a pull request for branch %q into branch %q already exists:\n%s",
+			ctx.HeadBranchLabel, ctx.BaseBranch, existingPR.URL)
+	}
```

Comment:
This may just be keeping status quo with current behavior but is there a reason why we wouldn't want to move these checks up to line 186, before the conditionals for `opts.WebMode` and `opts.Autofill || !opts.Interactive`?

Reply:
we would, I've since shuffled that around.